src/main/java/io/gatling/mojo/GatlingMojo.java

```diff
@@ -319,10 +322,17 @@ private void saveListOfNewRunDirectories(Exception exception) throws IOException
 
   private void writeExceptionIfExists(BufferedWriter writer, Exception exception) throws IOException {
     if (exception != null) {
-      writer.write(LAST_RUN_FILE_ERROR_LINE + " : " + exception.getClass() + " : " + exception.getMessage() + System.lineSeparator());
+      writer.write(LAST_RUN_FILE_ERROR_LINE + exception.getClass() + getRecursiveCauses(exception) + System.lineSeparator());
     }
   }
 
+  private String getRecursiveCauses(Throwable e) {
+    return stream(ExceptionUtils.getThrowables(e))
+            .filter(ex -> ex.getMessage() != null)
```

Comment:
Why do you filter out exceptions without a message? A simple NPE doesn't have one, and yet, we definitively want it.

Reply:
Misunderstood the original idea - fixed in new commit